packages/frontend/core/src/modules/quicksearch/impls/commands.ts

```diff
@@ -0,0 +1,210 @@
+    id: 'command:affine:collections',
+    label: {
+      key: 'com.affine.cmdk.affine.category.affine.collections',
+    },
+    score: 10,
+  },
+  'affine:settings': {
+    id: 'command:affine:settings',
+    label: { key: 'com.affine.cmdk.affine.category.affine.settings' },
+    score: 10,
+  },
+  'affine:updates': {
+    id: 'command:affine:updates',
+    label: { key: 'com.affine.cmdk.affine.category.affine.updates' },
+    score: 10,
+  },
+  'affine:help': {
+    id: 'command:affine:help',
+    label: { key: 'com.affine.cmdk.affine.category.affine.help' },
+    score: 10,
+  },
+  'editor:edgeless': {
+    id: 'command:editor:edgeless',
+    label: { key: 'com.affine.cmdk.affine.category.editor.edgeless' },
+    score: 10,
+  },
+  'editor:insert-object': {
+    id: 'command:editor:insert-object',
+    label: { key: 'com.affine.cmdk.affine.category.editor.insert-object' },
+    score: 10,
+  },
+  'editor:page': {
+    id: 'command:editor:page',
+    label: { key: 'com.affine.cmdk.affine.category.editor.page' },
+    score: 10,
+  },
+  'affine:results': {
+    id: 'command:affine:results',
+    label: { key: 'com.affine.cmdk.affine.category.results' },
+    score: 10,
+  },
+} satisfies Required<{
+  [key in CommandCategory]: QuickSearchGroup & { id: `command:${key}` };
+}>;
+
+function filterCommandByContext(
+  command: AffineCommand,
+  context: {
+    docMode: DocMode | undefined;
+  }
+) {
+  if (command.preconditionStrategy === PreconditionStrategy.Always) {
+    return true;
+  }
+  if (command.preconditionStrategy === PreconditionStrategy.InEdgeless) {
+    return context.docMode === 'edgeless';
+  }
+  if (command.preconditionStrategy === PreconditionStrategy.InPaper) {
+    return context.docMode === 'page';
+  }
+  if (command.preconditionStrategy === PreconditionStrategy.InPaperOrEdgeless) {
+    return !!context.docMode;
+  }
+  if (command.preconditionStrategy === PreconditionStrategy.Never) {
+    return false;
+  }
+  if (typeof command.preconditionStrategy === 'function') {
+    return command.preconditionStrategy();
+  }
+  return true;
+}
+
+function getAllCommand(context: { docMode: DocMode | undefined }) {
+  const commands = AffineCommandRegistry.getAll();
+  return commands.filter(command => {
+    return filterCommandByContext(command, context);
+  });
+}
+
+export class CommandsQuickSearchSession
+  extends Entity
+  implements QuickSearchSession<'commands', AffineCommand>
+{
+  constructor(private readonly contextService: GlobalContextService) {
+    super();
+  }
+
+  query$ = new LiveData('');
+
+  items$ = LiveData.computed(get => {
+    const query = get(this.query$);
+    const docMode =
+      get(this.contextService.globalContext.docMode.$) ?? undefined;
+    const commands = getAllCommand({ docMode });
+
+    const fuse = new Fuse(commands, {
+      keys: [{ name: 'label.title', weight: 2 }, 'label.subTitle'],
+      includeMatches: true,
+      includeScore: true,
+      threshold: 0.4,
+    });
```

Comment:
Is it better to wrap Fuse with a proper abstraction instead of accessing it directly?

Reply:
As a benefit, we can easily switch the underlying indexer engine and evaluate/compare the performance among them.

Reply:
This is only used to search for command and collection. Different search algorithms should be implemented according to different data. I will keep using fusejs here.